File: code/functions/callbacks.m
#import <Foundation/Foundation.h>

// Callbacks in Objective-C
typedef void (^CallbackBlock)(int value);

void callback(int value) {
    NSLog(@"Callback called with: %d", value);
}

void processWithCallback(void (*cb)(int), int data) {
    cb(data);
}

int main() {
    processWithCallback(callback, 42);
    return 0;
}

File: code/functions/callbacks.cpp
#include &lt;iostream&gt;
#include &lt;functional&gt;

// Callbacks in C++
void callback(int value) {
    std::cout &lt;&lt; "Callback called with: " &lt;&lt; value &lt;&lt; std::endl;
}

void process_with_callback(std::function&lt;void(int)&gt; cb, int data) {
    cb(data);
}

int main() {
    process_with_callback(callback, 42);
    return 0;
}